src/nimony/semdecls.nim

```diff
@@ -654,8 +654,94 @@ proc semProcImpl(c: var SemContext; it: var Item; kind: SymKind; pass: PassKind;
     producesVoid c, info, it.typ
   publish c, symId, declStart
 
+proc findMacroInvocs(c: SemContext; n: Cursor; kind: SymKind): seq[Cursor] =
+  # find all macro/template identifiers in pragmas to invoke them with parent proc definition
+  result = newSeq[Cursor]()
+  if kind in {ProcY, FuncY}:
```

Comment:
This should be done for all routine kinds, I think. But maybe not for templates, check what Nim 2.0 does.